contracts/yield-source/YearnV2YieldSource.sol

```diff
@@ -60,39 +67,66 @@ contract YearnV2YieldSource is IYieldSource, ERC20Upgradeable, OwnableUpgradeabl
+    /// @notice Mock Initializer to initialize implementations used by minimal proxies.
+    function freeze() public initializer {
+        //no-op
+    }
+
     /// @notice Initializes the yield source with
-    /// @param _vault YearnV2 Vault in which the Yield Source will deposit `token` to generate Yield
-    /// @param _token Underlying Token / Deposit Token
+    /// @dev `isContract()` will return false if an attacker is calling from a constructor https://ethereum.stackexchange.com/a/64340
+    /// @param _vault Yearn V2 Vault in which the Yield Source will deposit `token` to generate Yield
+    /// @param _decimals Number of decimals the shares (inherited ERC20) will have.  Same as underlying asset to ensure same ExchangeRates.
+    /// @param _symbol Token symbol for the underlying ERC20 shares (eg: yvysDAI).
+    /// @param _name Token name for the underlying ERC20 shares (eg: PoolTogether Yearn V2 Vault DAI Yield Source).
+    /// @param _owner Yearn V2 Vault Yield Source owner.
     function initialize(
         IYVaultV2 _vault,
-        IERC20Upgradeable _token
+        uint8 _decimals,
+        string calldata _symbol,
+        string calldata _name,
+        address _owner
     )
         public
         initializer
+        returns (bool)
     {
-        require(_vault.token() == address(_token), "YearnV2YieldSource:: incorrect vault");
-        require(_vault.activation() != uint256(0), "YearnV2YieldSource:: vault not initialized");
+        require(address(_vault) != address(0), "YearnV2YieldSource/vault-not-zero-address");
+        require(_vault.activation() != uint256(0), "YearnV2YieldSource/vault-not-initialized");
+
         // NOTE: Vaults from 0.3.2 to 0.3.4 have dips in shareValue
-        require(!areEqualStrings(_vault.apiVersion(), "0.3.2"), "YearnV2YieldSource:: vault not compatible");
-        require(!areEqualStrings(_vault.apiVersion(), "0.3.3"), "YearnV2YieldSource:: vault not compatible");
-        require(!areEqualStrings(_vault.apiVersion(), "0.3.4"), "YearnV2YieldSource:: vault not compatible");
+        string memory _vaultAPIVersion = _vault.apiVersion();
+
+        require(!_areEqualStrings(_vaultAPIVersion, "0.3.2"), "YearnV2YieldSource/vault-not-compatible");
+        require(!_areEqualStrings(_vaultAPIVersion, "0.3.3"), "YearnV2YieldSource/vault-not-compatible");
+        require(!_areEqualStrings(_vaultAPIVersion, "0.3.4"), "YearnV2YieldSource/vault-not-compatible");
 
         vault = _vault;
-        token = _token;
+        token = IERC20Upgradeable(_vault.token());
 
         __Ownable_init();
+        transferOwnership(_owner);
+
         __ReentrancyGuard_init();
 
-        _token.safeApprove(address(vault), type(uint256).max);
+        __ERC20_init(_name, _symbol);
+        require(_decimals > 0, "YearnV2YieldSource/decimals-not-greater-than-zero");
+        _setupDecimals(_decimals);
 
-        emit YieldSourceYearnV2Initialized(
+        token.safeApprove(address(_vault), type(uint256).max);
```

Comment:
We also need a function that can set to max.  Since we have an owner, the function can be restricted to just the owner.

Reply:
Done: https://github.com/pooltogether/pooltogether-yearnv2-yield-source/pull/8/commits/4d9c9978a15f5b51548872781e23a10e9081a605